subsys/storage/bm_storage/native_sim/bm_storage_native_sim.c

```diff
@@ -0,0 +1,128 @@
+/*
```

Comment:
It occurred to me that this backend perhaps should be integrated as part of the test and not placed here.